### DISCUSSION TITLE
What are the best settings, tweaks and parameter values for .generate and sam_kwargs?

### DISCUSSION THREAD

**Discussion by torrinworx** [opengeos/segment-geospatial#148] [Q&A] [answered]
I'm having a hard time finding the best values for all these parameters in .generate and sam_kwargs, does anyone have any default values that work really well for auto segmenting satellite imagery?

**Comment by giswqs** [comment, marked as answer]
The model paraters are from the original SAM model. The default values should work fine in most cases. You can fine-tune the parameters, but I am not aware of any general guidelines for this. 
https://github.com/facebookresearch/segment-anything/blob/main/segment_anything/automatic_mask_generator.py#L36

**Reply by torrinworx** [reply]
Ah thank you, I've given the default values a try, however they don't seem to segment the entire image, there are still patches that haven't been masked out. I'm just curious if anyone has gotten better results with other values.

I appreciate the quick response!

**Reply by giswqs** [reply]
Try setting `foreground=False`

https://samgeo.gishub.org/samgeo/#samgeo.samgeo.SamGeo.generate

**Reply by torrinworx** [reply]
Brilliant, thank you @giswqs!
